src/main/java/com/conveyal/datatools/manager/models/FeedSource.java

```diff
@@ -538,4 +544,88 @@ public FeedSource clone () throws CloneNotSupportedException {
         return (FeedSource) super.clone();
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+        FeedSource that = (FeedSource) o;
+        return id.equals(that.id);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id);
+    }
+
+    /**
+     * Cancels all existing feed expiration notifications
+     */
+    public void cancelExpirationNotifications() {
+        for (Future expirationNotification : scheduledNotificationExpirations.get(this)) {
+            expirationNotification.cancel(true);
+        }
+        scheduledNotificationExpirations.removeAll(this);
+    }
+
+    public void scheduleExpirationNotifications () {
+        // cancel existing expiration notifications
+        cancelExpirationNotifications();
+
+        FeedVersion latest = retrieveLatest();
+
+        if (
+            latest != null &&
+            latest.validationResult != null &&
+                latest.validationResult.lastCalendarDate.isAfter(LocalDate.now())
+        ) {
+            // get parent project
+            Project parentProject = this.retrieveProject();
+
+            if (parentProject == null) {
+                // parent project has been deleted, but feed source/version have not
+                // abort the setting up of the notification and figure out why the database has been
+                // allowed to devolve to this state
+                return;
+            }
+
+            // get the timezone from the parent project
+            ZoneId timezone = getTimezone(parentProject.defaultTimeZone);
+
+            // calculate feed expiration time from last service date
+            long expirationEpochSeconds = latest
+                .validationResult
+                .lastCalendarDate
+                .atTime(0, 0)
```

Comment:
9am is way too late in the day because some people could start working before that.  Maybe 4am is best since it wouldn't usually conflict with other items running on the server.  I think this would be a good thing to be one of the first things someone sees in their inbox perhaps along with a nightly status report / summary.